tests/src/Integration/Action/SystemMailToUsersOfRoleTest.php

```diff
@@ -0,0 +1,239 @@
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $test_email='admin@example.com';
+
+    // Create two user roles and one user for each.
+    $uid = 1;
+    foreach (['administrator', 'editor'] as $role_name) {
+
+      // Mock the role entity.
+      $role_mock = $this->prophesize(RoleInterface::class);
+      $role_mock->id()
+        ->willReturn($role_name);
+      $this->roles[] = $role_mock->reveal();
+
+      // Mock the user entity.
+      $user_mock = $this->prophesizeEntity(UserInterface::class);
+      $user_mock->id()->willReturn($uid);
+
+      $user_entity_type = $this->prophesize(EntityTypeInterface::class);
+      $user_mock->getEntityType()
+        ->willReturn($user_entity_type->reveal());
+
+      $user_mock->getPreferredLangcode()
+        ->willReturn(LanguageInterface::LANGCODE_SITE_DEFAULT);
+
+      $user_mock->getEmail()
+        ->willReturn($test_email);
+
+      $this->users[$uid] = $user_mock->reveal();
+      $uid++;
+    }
+
+    $this->logger = $this->prophesize(LoggerInterface::class);
+    $logger_factory = $this->prophesize(LoggerChannelFactoryInterface::class);
+    $logger_factory->get('rules')->willReturn($this->logger->reveal());
+
+    $this->mailManager = $this->prophesize(MailManagerInterface::class);
+    $this->userStorage = $this->prophesize(UserStorageInterface::class);
+
+    // Mocked entityManager.
+    $this->entityManager = $this->prophesize(EntityManagerInterface::class);
+
+    $this->entityManager->getStorage('user')
+      ->willReturn($this->userStorage->reveal());
+
+    $this->entityManager->createInstance()
+      ->willReturn($this->userStorage->reveal());
+
+    // Prepare a user entity type instance.
+    $entity_type = reset($this->users)->getEntityType();
+    $this->entityManager->getDefinitions()
+      ->willReturn($entity_type);
+
+    // @todo this is wrong, the logger is no factory.
+    $this->container->set('logger.factory', $logger_factory->reveal());
+    $this->container->set('plugin.manager.mail', $this->mailManager->reveal());
+    $config = [
+      'system.site' => [
+        'mail' => $test_email,
+      ],
+    ];
+    $this->container->set('config.factory', $this->getConfigFactoryStub($config));
+    $this->container->set('entity.manager', $this->entityManager->reveal());
+
+    $this->action = $this->actionManager->createInstance('rules_mail_to_users_of_role');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary
+   */
+  public function testSummary() {
+    $this->assertEquals('Mail to users of a role', $this->action->summary());
+  }
+
+  /**
+   * Tests sending a mail to one or two roles.
+   *
+   * @param int $call_number
+   *   The number of emails that should be sent.
+   * @param int $role_number
+   *   The number of roles (1 or 2).
+   *
+   * @dataProvider providerSendMailToRole
+   *
+   * @covers ::execute
+   */
+  public function testSendMailToRole($call_number, $role_number) {
```

Comment:
Instead of passing in $role_number here you should pass in the actual role object.